js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNodeStatusContent.tsx

```diff
@@ -0,0 +1,382 @@
+import {Body, Colors, Icon, Spinner} from '@dagster-io/ui-components';
+import React from 'react';
+import {Link} from 'react-router-dom';
+
+import {
+  StyleForAssetPartitionStatus,
+  partitionCountString,
+} from '../assets/AssetNodePartitionCounts';
+import {AssetPartitionStatusDot} from '../assets/AssetPartitionList';
+import {AssetPartitionStatus} from '../assets/AssetPartitionStatus';
+import {OverdueLineagePopover, isAssetOverdue} from '../assets/OverdueTag';
+import {assetDetailsPathForKey} from '../assets/assetDetailsPathForKey';
+import {
+  AssetCheckExecutionResolvedStatus,
+  AssetCheckSeverity,
+  AssetKeyInput,
+} from '../graphql/types';
+import {TimestampDisplay} from '../schedules/TimestampDisplay';
+
+import {AssetLatestRunSpinner, AssetRunLink} from './AssetRunLinking';
+import {LiveDataForNode, stepKeyForAsset} from './Utils';
+
+export enum StatusCase {
+  LOADING = 'LOADING',
+  SOURCE_OBSERVING = 'SOURCE_OBSERVING',
+  SOURCE_OBSERVED = 'SOURCE_OBSERVED',
+  SOURCE_NEVER_OBSERVED = 'SOURCE_NEVER_OBSERVED',
+  SOURCE_NO_STATE = 'SOURCE_NO_STATE',
+  MATERIALIZING = 'MATERIALIZING',
+  LATE_OR_FAILED = 'LATE_OR_FAILED',
+  NEVER_MATERIALIZED = 'NEVER_MATERIALIZED',
+  MATERIALIZED = 'MATERIALIZED',
+  PARTITIONS_FAILED = 'PARTITIONS_FAILED',
+  PARTITIONS_MISSING = 'PARTITIONS_MISSING',
+  PARTITIONS_MATERIALIZED = 'PARTITIONS_MATERIALIZED',
+}
+
+type AssetNodeStatusContent = {
+  case: StatusCase;
+  background: string;
+  border: string;
+  content: React.ReactNode;
+
+  numPartitions?: number;
```

Comment:
rather than typing these out as all nullable here we should use ReturnType here so that we get the correct types as nullable/non-nullable depending on the status case

Reply:
Oh interesting will try that! I wonder if it can correctly build a union type? That'd actually be really nice

Reply:
Ahh actually it looks like I only use this type to type the various responses to make sure they all contain the required keys -- if I switch to ReturnType and remove the type annotations inside the `buildAssetNodeStatusContent` method itself, there's nothing left 😂

I think it's safe to just remove this type.